examples/smartcore-ml/README.md

```diff
@@ -0,0 +1,119 @@
+
+Regression:  Linear Regression, Lasso, Ridge, and Elastic Net.
+
+Clutering:  K-Means, DBSCAN.
+
+Dimensionality Reduction:  PCA.
+
+Matrix Decomposition:  SVD, Eigenvalue Decomposition, QR, LU.
+
+The following folders are not part of the standard RISC Zero starter template and should be added if the user wishes to replicate the layout of this template:  res/ml-model, res/input-data, and train-model.  ml-model contains a json file of a trained decision tree model and input-data contains a json file of sample input data.  You can train and export your model along with any sample data in train-model/src/main.rs (note that this is excluded from the zkvm workspace).  
+
+res/ml-model and res/input-data come preloaded with a trained decision tree classifier.  You can replace the code in train-model to train and export any of the supported models listed above.
+
+## Model Types
+
+It is important to specificy the generic types for each model so that they can be succesfully deserialized.  The input data for a trained model is typically formatted as a DenseMatrix, so be sure to add the following import: 
+`use smartcore::linalg::basic::matrix::DenseMatrix;`
+when using KNN, be sure to add the following import:
+`use smartcore::metrics::distance::euclidian::Euclidian;`
+A list of the supported models with definitions for their respective types can be found below.  Note that {float} is a placeholder for either f32 or f64:
+
+```
+let model: DecisionTreeClassifier<{float}, u32, DenseMatrix<{float}>, Vec<u32>> = trained_model;
+```
+
+```
+let model: RandomForestClassifier<{float}, u8, DenseMatrix<{float}>, Vec<u8>> = trained_model;
+```
+
+```
+let model: GaussianNB<{float}, _, DenseMatrix<{float}>, _> = trained_model;
+```
+
+```
+let model: LogisticRegression<{flaot}, u32, DenseMatrix<{float}>, Vec<u32>> = trained_model;
+```
+
+```
+let model: LinearRegression< {float}, {float}, DenseMatrix<{float}>, Vec<{float}>> = trained_model;
+```
+
+```
+let model: KNNClassifier< {float}, u32, DenseMatrix<{float}>, Vec<u32>, Euclidian<{float}>> = trained_model;
+```
+
+```
+let model: KMeans<{float}, u8, DenseMatrix<{float}>, Vec<u8>> = trained_model;
+```
+
+```
+let model: PCA<{float}, DenseMatrix<{float}>> = trained_model;
+```
+
+NOTE:  For Support Vector Machine classifiers and regression models, you MUST use the following smartcore fork and import into you cargo.toml file:
+
+smartcore = { git = "https://github.com/Roee-87/smartcore.git", features = ["serde"]}
+
+Deserialization of SVC and SVR does not includes that SVCParametersmake field.  It must be added back into the model struct manually after deserialization.  The SVC and SVR fork have ammended the visbility of the model struct, making the parameters field public and thereby allowing the parameters field to be directly inserted into the model struct after dserialization.  You must insert the same parameters that were used when training the model.
```

Comment:
Some proofreading suggestions. I don't know I corrected correctly to "The SVC and SVR forks have" ... perhaps it's "The SVC and SVR fork has"?
```suggestion
Deserialization of SVC and SVR does not includes that `SVCParametersmake` field.  It must be added back into the model struct manually after deserialization.  The SVC and SVR forks have amended the visbility of the model `struct`, making the parameters field public and thereby allowing the parameters field to be directly inserted into the model struct after deserialization.  You must insert the same parameters that were used when training the model.
```